language/move-analyzer/src/symbols.rs

```diff
@@ -0,0 +1,2164 @@
+            // difference from the IDE perspective as the cursor goes to the beginning anyway (at
+            // least in VSCode).
+            let range = Range {
+                start: u.def_loc.start,
+                end: u.def_loc.start,
+            };
+            let path = symbols.file_name_mapping.get(&u.def_loc.fhash).unwrap();
+            let loc = Location {
+                uri: Url::from_file_path(path.as_str()).unwrap(),
+                range,
+            };
+            Some(serde_json::to_value(loc).unwrap())
+        },
+    );
+}
+
+/// Handles go-to-references request of the language server
+pub fn on_references_request(context: &Context, request: &Request, symbols: &Symbols) {
+    let parameters = serde_json::from_value::<ReferenceParams>(request.params.clone())
+        .expect("could not deserialize references request");
+
+    let fpath = parameters.text_document_position.text_document.uri.path();
+    let loc = parameters.text_document_position.position;
+    let line = loc.line;
+    let col = loc.character;
+    let include_decl = parameters.context.include_declaration;
+
+    on_use_request(
+        context,
+        symbols,
+        fpath,
+        line,
+        col,
+        request.id.clone(),
+        |u| match symbols.references.get(&u.def_loc) {
+            Some(s) => {
+                let mut locs = vec![];
+                for ref_loc in s {
+                    if include_decl
+                        || !(u.def_loc.start == ref_loc.start && u.def_loc.fhash == ref_loc.fhash)
+                    {
+                        let end_pos = Position {
+                            line: ref_loc.start.line,
+                            character: ref_loc.col_end,
+                        };
+                        let range = Range {
+                            start: ref_loc.start,
+                            end: end_pos,
+                        };
+                        let path = symbols.file_name_mapping.get(&ref_loc.fhash).unwrap();
+                        locs.push(Location {
+                            uri: Url::from_file_path(path.as_str()).unwrap(),
+                            range,
+                        });
+                    }
+                }
+                if locs.is_empty() {
+                    Some(serde_json::to_value(Option::<lsp_types::Location>::None).unwrap())
+                } else {
+                    Some(serde_json::to_value(locs).unwrap())
+                }
+            }
+            None => Some(serde_json::to_value(Option::<lsp_types::Location>::None).unwrap()),
+        },
+    );
+}
+
+/// Helper function to handle language server queries related to identifier uses
+pub fn on_use_request(
+    context: &Context,
+    symbols: &Symbols,
+    use_fpath: &str,
+    use_line: u32,
+    use_col: u32,
+    id: RequestId,
+    use_def_action: impl Fn(&UseDef) -> Option<serde_json::Value>,
+) {
+    let mut result = Option::<serde_json::Value>::None;
```

Comment:
Second use of `Option::None` I've seen. Whats the issue you ran into when you just used `None`?

Reply:
No error... Might have been a copypasta from the use in the previous method when you do need to specify the type.